### PR TITLE
feat: add user guide

### PR DESCRIPTION
**Pull request by yanzhihong23**
closes #

### CODE REVIEW

**Comment by guanbinrui** [comment]
@DimensionDev/qa

**Comment by wenluomask** [comment]
1. Will mess up if other Twitter home pages are opening;
2. User guide is suppose to show only once, currently it will show up once we connect Twitter account;
3. The first "Find your user name" dialog‘s background should be white for dark mode;
Screen Shot 2021-09-28 at 10 26 20 PM
4. Change some text for user guide;(Wait for confirmation)
5. Twitter username is missing in the first step;
Screen Shot 2021-09-28 at 10 19 20 PM

**Comment by yanzhihong23** [comment]
@wenluomask updated, please check again.

**Comment by wenluomask** [comment]
Checked.

File: packages/maskbook/src/components/GuideStep/index.tsx [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+    const childrenRef = useRef<HTMLElement>()
+    const [clientRect, setClientRect] = useState<any>({})
+    const [open, setOpen] = useState(false)
+    const ui = activatedSocialNetworkUI
+    const lastStepRef = currentSetupGuideStatus[ui.networkIdentifier]
+    const lastStep = useValueRef(lastStepRef)
+
+    useEffect(() => {
+        setOpen(+lastStep === step)
+    }, [lastStep])
```
Comment (guanbinrui) [inline packages/maskbook/src/components/GuideStep/index.tsx:165]: Just for double confirmation: it's on purpose to exclude `step` from the deps list.

**Review by guanbinrui — APPROVED** [review]

File: packages/maskbook/src/components/InjectedComponents/SetupGuide.tsx [hunk 2]
```diff
@@ -37,18 +36,21 @@ export enum SetupGuideStep {
     FindUsername = 'find-username',
     SayHelloWorld = 'say-hello-world',
 }
+
+const userGuideCompleted = localStorage.getItem('userGuideCompleted')
```
Comment (Jack-Works) [inline packages/maskbook/src/components/InjectedComponents/SetupGuide.tsx:40]: @guanbinrui you're storing it in the SNS's local storage. don't do it.

Comment (yanzhihong23) [reply packages/maskbook/src/components/InjectedComponents/SetupGuide.tsx:40]: @Jack-Works #4536

**Pull request merged**